python/tests/test_ibd.py

```diff
@@ -0,0 +1,821 @@
+            (0, 3): [ibd.Segment(0.3, 1, 4)],
+            (1, 2): [ibd.Segment(0.3, 1, 5)],
+            (1, 3): [ibd.Segment(0.3, 1, 4)],
+            (2, 3): [ibd.Segment(0.3, 1, 5)],
+        }
+        ibd_segs = convert_dict_of_segmentlists(ibd_segs)
+        assert ibd_is_equal(ibd_segs, true_segs)
+
+
+class TestIbdInternalSamples(unittest.TestCase):
+    #
+    #
+    #      3
+    #     / \
+    #    /   2
+    #   /     \
+    #  0      (1)
+    nodes = io.StringIO(
+        """\
+    id      is_sample   time
+    0       1           0
+    1       0           0
+    2       1           1
+    3       0           2
+    """
+    )
+    edges = io.StringIO(
+        """\
+    left    right   parent  child
+    0.0     1.0     2       1
+    0.0     1.0     3       0
+    0.0     1.0     3       2
+    """
+    )
+    ts = tskit.load_text(nodes=nodes, edges=edges, strict=False)
+
+    def test_defaults(self):
+        ts = self.ts
+        ibd_f = ibd.IbdFinder(ts)
+        ibd_segs = ibd_f.find_ibd_segments()
+        ibd_segs = convert_dict_of_segmentlists(ibd_segs)
+        true_segs = {
+            (0, 2): [ibd.Segment(0, 1, 3)],
+        }
+        assert ibd_is_equal(ibd_segs, true_segs)
+
+
+class TestIbdRandomExamples(unittest.TestCase):
+    """
+    Randomly generated test cases.
+    """
+
+    # Infinite sites, Hudson model.
+    @unittest.skipIf(
+        version.parse(msprime.__version__) < version.parse("0.7.5"),
+        "Needs finite sites simulations to work.",
+    )
+    def test_random_example1(self):
+        ts = msprime.simulate(sample_size=10, recombination_rate=0.5, random_seed=2)
+        verify_equal_ibd(ts)
+
+    @unittest.skipIf(
+        version.parse(msprime.__version__) < version.parse("0.7.5"),
+        "Needs finite sites simulations to work.",
+    )
+    def test_random_example2(self):
+        ts = msprime.simulate(sample_size=10, recombination_rate=0.5, random_seed=23)
+        verify_equal_ibd(ts)
+
+    @unittest.skipIf(
+        version.parse(msprime.__version__) < version.parse("0.7.5"),
+        "Needs finite sites simulations to work.",
+    )
+    def test_random_example3(self):
+        ts = msprime.simulate(sample_size=10, recombination_rate=0.5, random_seed=232)
+        verify_equal_ibd(ts)
+
+    @unittest.skipIf(
+        version.parse(msprime.__version__) < version.parse("0.7.5"),
+        "Needs finite sites simulations to work.",
+    )
+    def test_random_example4(self):
+        ts = msprime.simulate(sample_size=10, recombination_rate=0.3, random_seed=726)
+        verify_equal_ibd(ts)
+
+    # Finite sites
+    def sim_finite_sites(self, random_seed, dtwf=False):
+        seq_length = int(1e5)
+        positions = random.sample(range(1, seq_length), 98) + [0, seq_length]
+        positions.sort()
+        rates = [random.uniform(1e-9, 1e-7) for _ in range(100)]
+        r_map = msprime.RecombinationMap(
+            positions=positions, rates=rates, discrete=True
```

Comment:
It would be simpler to use the shipping APIs for msprime than to do all this skipping (plus, we don't want to skip the tests - otherwise how do we know they are doing anything?). Use ``num_loci=seq_length`` to achieve the same result here. It is *definitely* possible to have discrete recombination breakpoints with msprime < 1.0